crates/ruff/src/rules/flake8_newspaper_style/rules/check_function_calls.rs

```diff
@@ -0,0 +1,69 @@
+use rustpython_parser::ast::{Expr, ExprKind, Located};
+
+use ruff_diagnostics::{Diagnostic, Violation};
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_ast::types::{Binding, BindingKind, FunctionDef, Range, Scope, ScopeKind};
+
+use crate::checkers::ast::Checker;
+use crate::registry::AsRule;
+
+#[violation]
+pub struct FuncDefinedAbove {
+    pub function_name: String,
+    pub function_def_line: usize,
+}
+
+impl Violation for FuncDefinedAbove {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        format!(
+            "newspaper style: function {} defined in line {} should be moved down.",
+            self.function_name, self.function_def_line
+        )
+    }
+}
+
+fn check_if_func_defined_above(checker: &mut Checker, func: &Expr) -> Option<Diagnostic> {
+    // println!("{scope:?}", scope = checker.ctx.current_scope_parent());
+    let kind = &checker.ctx.current_scope().kind;
+    if let ScopeKind::Function(FunctionDef {
+        name: caller_name, ..
+    }) = kind
+    {
+        let Located { node, .. } = func;
+        if let ExprKind::Attribute {
+            attr: callee_name, ..
+        }
+        | ExprKind::Name {
+            id: callee_name, ..
+        } = node
+        {
+            if let Some(Scope { bindings, .. }, ..) = checker.ctx.current_scope_parent() {
+                if bindings[caller_name] > *bindings.get(&callee_name[..]).unwrap_or(&usize::MAX) {
+                    if let Some(Binding { kind, range, .. }) = checker.ctx.find_binding(callee_name)
+                    {
```

Comment:
Nit: `checker.ctx.bindngs[*index]` (you already have the index)

Reply:
Where? I can't find it 🙈